pages/api/service/index.ts

```diff
@@ -230,17 +124,61 @@ export default async function handler(
           .skip(+skip)
           .populate('domain')
           .populate('street')
-        const domainFilter = await getDomainFilter()
-        const addressFilter = await getAddressFilter()
-        const yearFilter = await getYearFilter()
-        const monthFilter = await getMonthFilter()
+
+        const distinctStreets = await Service.distinct('street', options)
+        const distinctDomains = await Service.distinct('domain', options)
+
+        const [relatedDomains, relatedStreets, relatedDates] =
+          await Promise.all([
+            Domain.find({ _id: { $in: distinctDomains } }).select([
+              'name',
+              '_id',
+            ]),
+            Street.find({ _id: { $in: distinctStreets } }).select([
+              'address',
+              'city',
+              '_id',
+            ]),
+          ])
```

Comment:
Your `Promise.all` has 2 promises inside it, but your response array expects 3. Move rest of your requests inside this `Promise.all` to increase performace.

```
const [relatedDomains, relatedStreets, distinctYears, distinctMonths] =
  await Promise.all([
    Domain.find({ _id: { $in: distinctDomains } }).select(['name', '_id']),

    Street.find({ _id: { $in: distinctStreets } }).select(['address', 'city', '_id']),

    Service.distinct('date', options).then((dates) => [
      ...new Set(dates.map((date) => new Date(date).getFullYear())),
    ]),

    Service.aggregate([
      { $match: options },
      { $group: { _id: { month: { $month: '$date' } } } },
      { $sort: { '_id.month': 1 } },
    ]).then((results) => [
      ...new Set(results.map((result) => result._id.month)),
    ]),
  ])
```